File: Tutorial1DaggerBasics2/src/main/java/com/example/tutorial1basics3/MyApplication.java
package com.example.tutorial1basics3;

import android.app.Application;

import com.example.tutorial1basics3.di.AppContextComponent;
import com.example.tutorial1basics3.di.AppContextModule;
import com.example.tutorial1basics3.di.DaggerAppContextComponent;

public class MyApplication extends Application {

    private AppContextComponent mAppContextComponent;

    @Override
    public void onCreate() {
        super.onCreate();

        mAppContextComponent = DaggerAppContextComponent
                .builder()
                .appContextModule(new AppContextModule(this))
                .build();
    }


    public AppContextComponent getAppContextComponent() {
        return mAppContextComponent;
    }
}

File: Tutorial1-2DaggerBasics2/src/main/java/com/example/tutorial1basics3/MyApplication.kt
package com.example.tutorial1basics3

import android.app.Application
import com.example.tutorial1basics3.di.AppContextComponent
import com.example.tutorial1basics3.di.AppContextModule
import com.example.tutorial1basics3.di.DaggerAppContextComponent

class MyApplication : Application() {

    lateinit var appContextComponent: AppContextComponent
        private set

    override fun onCreate() {
        super.onCreate()
        appContextComponent = DaggerAppContextComponent
                .builder()
                .appContextModule(AppContextModule(this))
                .build()
    }

}